### PR TITLE
Create multiple sender workers using one forwarder

### PR DESCRIPTION
**Pull request by maorlx**

### CODE REVIEW

File: pkg/cluster_agent/main.py [hunk 1]
```diff
@@ -73,7 +74,7 @@ async def run():
             logging.error(format_exc())
             logging.info("Agent is exiting due to an unexpected error")
             _cancel_tasks(tasks)
-            epsagon_client.close()
+            await psagon_client.close()
```
Comment (Alon-Katz) [inline pkg/cluster_agent/main.py]: Typo psagon_client

File: pkg/cluster_agent/main.py [hunk 2]
```diff
@@ -38,22 +38,23 @@ def _cancel_tasks(tasks):
...
     events_sender = EventsSender(epsagon_client, COLLECTOR_URL)
     cluster_discovery = ClusterDiscovery(events_manager.write_event)
-    forwarders = [
-        Forwarder(events_manager, events_sender)
-        for _ in range(DEFAULT_COLLECTOR_WORKERS_COUNT)
-    ]
+    forwarder = Forwarder(
+        events_manager,
+        events_sender,
+        max_workers=DEFAULT_COLLECTOR_WORKERS_COUNT
```
Comment (Alon-Katz) [inline pkg/cluster_agent/main.py]: Shouldn't we allow for max workers and max events to be controlled outside with env params?
Also if we already have a default value for max workers in Forwarder class, why do we another default value?

File: pkg/cluster_agent/forwarder.py [hunk 3]
```diff
@@ -12,20 +12,37 @@ class Forwarder:
...
         :param events_manager: used to read from events
         :param events_sender: used to send read events to
+        :param max_workers: to forward read events
+        :param max_events_to_read: to read from the events_manager
         """
         self.events_manager = events_manager
         self.events_sender = events_sender
+        if max_workers < 1:
+            raise ValueError("Invalid workers count value, must be > 0")
+        self.max_workers_count: int = max_workers
```
Comment (Alon-Katz) [inline pkg/cluster_agent/forwarder.py:35]: Do we need the int definition here when the function param already mentions that max_workers is int?
if yes, then we should also add it to max_events_to_read for example

**Review by Alon-Katz — CHANGES_REQUESTED** [review]

File: pkg/cluster_agent/forwarder.py (hunk 3 above)
Comment (maorlx) [reply pkg/cluster_agent/forwarder.py:35]: not mandatory, but will add for max_events

File: pkg/cluster_agent/main.py (hunk 2 above)
Comment (maorlx) [reply pkg/cluster_agent/main.py]: will change the forwarder default to be as the main one (and will remove the one at main)
Regarding the env var, I am not sure we actually would like to allow this. Lets talk about it

**Review by Alon-Katz — APPROVED** [review]

**Pull request merged**